src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinInetProxyHelper.cs

```diff
@@ -178,5 +153,97 @@ public bool ManualSettingsOnly
 
             return useProxy;
         }
+
+        private void DetectScriptLocation()
+        {
+            if (_scriptDetectionFailed || AutoConfigUrl != null)
+            {
+                return;
+            }
+
+            WinHttpTraceHelper.Trace("WinInetProxyHelper.DetectScriptLocation: Attempting discovery PROXY_AUTO_DETECT_TYPE_DHCP.");
+            AutoConfigUrl = SafeDetectAutoProxyUrl(Interop.WinHttp.AutoDetectType.Dhcp);
+
+            if (AutoConfigUrl == null)
+            {
+                WinHttpTraceHelper.Trace("WinInetProxyHelper.DetectScriptLocation: Attempting discovery AUTO_DETECT_TYPE_DNS_A.");
+                AutoConfigUrl = SafeDetectAutoProxyUrl(Interop.WinHttp.AutoDetectType.DnsA);
+            }
+            else
+            {
+                // Auto-detect succeed.
+                return;
+            }
+
+            if (AutoConfigUrl == null)
+            {
+                WinHttpTraceHelper.Trace("WinInetProxyHelper.DetectScriptLocation: Auto discovery failed.");
+                AutoDetect = false;
+                _scriptDetectionFailed = true;
+            }
+        }
+
+        private string SafeDetectAutoProxyUrl(Interop.WinHttp.AutoDetectType discoveryMethod)
```

Comment:
Why use 'Safe' as the prefix of this method name?  Maybe call this method something like "GetAutoProxyUrl".

Reply:
Will do.